webapp-django/crashstats/crashstats/tests/test_models.py

```diff
@@ -34,6 +33,134 @@ def json(self):
         return self.raw
 
 
+class TestGraphicsDevices(DjangoTestCase):
+    def setUp(self):
+        super(TestGraphicsDevices, self).setUp()
+        # thanks to crashstats.settings.test
+        assert settings.CACHE_IMPLEMENTATION_FETCHES
+        cache.clear()
+
+    def tearDown(self):
+        super(TestGraphicsDevices, self).tearDown()
+
+    def test_get_pairs(self):
+        """Test get_pairs() works correctly
+
+        The GraphicsDevice.get_pairs() lets you make a bunch of requests at
+        the same time. It's more performant since it caches results.
+
+        """
+        models.GraphicsDevice.objects.create(
+            vendor_hex='vhex3',
+            vendor_name='V 3',
+            adapter_hex='ahex3',
+            adapter_name='A 3'
+        )
+        models.GraphicsDevice.objects.create(
+            vendor_hex='vhex2',
+            vendor_name='V 2',
+            adapter_hex='ahex2',
+            adapter_name='A 2',
+        )
+        models.GraphicsDevice.objects.create(
+            vendor_hex='vhex1',
+            vendor_name='V 1',
+            adapter_hex='ahex1',
+            adapter_name='A 1',
+        )
+
+        r = models.GraphicsDevice.objects.get_pairs(
+            ['ahex1', 'ahex2'],
+            ['vhex1', 'vhex2'],
+        )
+        expected = {
+            ('ahex1', 'vhex1'): ('A 1', 'V 1'),
+            ('ahex2', 'vhex2'): ('A 2', 'V 2'),
+        }
+        assert r == expected
+
+        r = models.GraphicsDevice.objects.get_pairs(
+            ['ahex2', 'ahex3'],
+            ['vhex2', 'vhex3'],
+        )
+        assert len(r) == 2
+        expected = {
+            ('ahex2', 'vhex2'): ('A 2', 'V 2'),
+            ('ahex3', 'vhex3'): ('A 3', 'V 3'),
+        }
+        assert r == expected
+
+        # FIXME(willkg): verify caching is working
```

Comment:
This was part of `TestModels` and I broke it out into its own class. Otherwise, the code is identical.